### PR TITLE
BIND: adding bind support to sms parser

### LINKED ISSUE #49: Update SMS parser to bring inline with Stardog docs
Pending docs changes are here: https://github.com/stardog-union/web/pull/1222 .

Marking this as a bug because it already creates some issues in "the real world" in Stardog Studio (the lines marked as invalid are actually valid):

image

- Add `IRIFunction` to bind
- Add `StrDt` to bind
- Add `StrLang` to bind
- Add CSV as valid `FromClause`

### PR DESCRIPTION
**Pull request by cchaffatt**
# Bind support for sms parser
* PR coding with @anneeb 
* line 854 => [support doc](https://github.com/stardog-union/web/pull/1222/files)
* fix #49

### CODE REVIEW

File: src/sms/SmsParser.ts [hunk 1]
```diff
@@ -141,41 +123,7 @@ export class SmsParser extends Parser {
   //
   // Dupes from Sparql
   //
```
[the remark this answers is not in the document]
Comment (cchaffatt) [reply src/sms/SmsParser.ts:125]: They were duplicates until we changed `RULE` to `OVERRIDE_RULE`. So i guess maybe the comment can be changed or deleted?

[the remark this answers is not in the document]
Comment (cchaffatt) [reply src/sms/SmsParser.ts:125]: Agreed, we should completely fix the ticket in this PR

[the remark this answers is not in the document]
Comment (anneeb) [reply src/sms/SmsParser.ts:125]: I think this comment should be something more descriptive about the fact that these are`overrides for stardog sparql parser back to original base sparql parser behavior

**Comment by anneeb** [comment]
> This is looking good -- just two small comments inline, and then I _could_ approve it, though I have to note that it won't fully fix #49 as it purports to. This is because it doesn't support the "FROM CSV" SMS2 syntax that is now supported in Stardog. The documentation for this admittedly isn't great, and it's _barely_ mentioned in the Stardog docs mentioned in #49 (it's mentioned only as "CSV" in the "Mapping" table here: https://www.stardog.com/docs/#_mapping), but it is something we want to address. Basically, the `FROM` clause can now be either `FROM JSON`, `FROM SQL`, `FROM GRAPHQL`, or `FROM CSV`. In the `FROM CSV` case, it can be followed only by empty braces; the braces can also be omitted.
> 
> We can either add support for that into this PR, or open a separate PR adding support for that. In the latter case, we'd have to mark this as not fixing #49 (so that #49 doesn't auto-close).

We should also add tests for the CSV clause if we do add it here

**Pull request merged**